tests/unit/pypyr/stepsrunner_test.py

```diff
@@ -1,12 +1,34 @@
 """stepsrunner.py unit tests."""
+from copy import deepcopy
 import logging
 import pytest
-from unittest.mock import call, patch
+from unittest.mock import call, patch, MagicMock
 from pypyr.context import Context
 from pypyr.dsl import Step
 from pypyr.errors import ContextError
 import pypyr.stepsrunner
 
+
+class DeepCopyMagicMock(MagicMock):
```

Comment:
copied from dsl_test, probably should move both to a separate file..

Reply:
yes, it's probably about time for a ./tests/common/utils.py or something of the sort.